src/controllers/sqlDocumentService.ts

```diff
@@ -182,17 +183,20 @@ export default class SqlDocumentService implements vscode.Disposable {
 
         if (doc && doc.uri) {
             this._lastOpenedUri = getUriKey(doc.uri);
-
-            // pre-opened tabs won't trigger onDidChangeActiveTextEditor, so set _previousActiveEditor here
-            this._previousActiveDocument =
-                doc.languageId === Constants.languageId ? doc : undefined;
         }
     }
 
     public async onDidChangeActiveTextEditor(editor: vscode.TextEditor | undefined): Promise<void> {
-        if (editor?.document) {
-            this._previousActiveDocument =
-                editor.document.languageId === Constants.languageId ? editor.document : undefined;
+        if (!editor?.document) {
+            return;
+        }
+
+        const activeDocumentUri = getUriKey(editor.document.uri);
+        const activeConnection =
+            await this._connectionMgr?.getConnectionInfoFromUri(activeDocumentUri);
+
+        if (activeConnection) {
+            this._lastActiveConnectionInfo = activeConnection;
```

Comment:
Is there any problem/risk of the active connection getting changed, with it getting set by reference?  Should this be a copy instead, or just an ID?